pandas/core/groupby/groupby.py

```diff
@@ -1837,14 +1837,26 @@ def mean(
         Name: B, dtype: float64
         """
 
+        if not (
+            type(self.obj).mean is Series.mean or type(self.obj).mean is DataFrame.mean
+        ):
```

Comment:
What about a Boolean flag as a class attribute on Series / DataFrame that subclasses can override if they want to use their methods? Defaults to False.

Reply:
> Does .groupby(...).agg(["sum", "mean"]) work with this?

No: 

```
>>> udf.groupby('group').agg(['sum', 'mean'])
              a                   b
            sum      mean       sum      mean
group
1      0.322530  0.161265  0.769053  0.384527
2      1.714631  0.857315  0.407676  0.203838
```
Using my example where the mean should always be 1.

Reply:
However transform() does work:
```
>>> udf.groupby('group').transform('mean')
   a  b
0  1  1
1  1  1
2  1  1
3  1  1
```

Reply:
Thanks for checking. In my opinion that isn't necessarily a blocker for this, but we are expanding the behavior here and introducing new bugs that should get fixed.

If we are to move forward with some form of implementation for this, I think the test coverage here should be expanded to include at least agg, apply, and transform (even if they are currently producing incorrect results).